pkg/objstore/s3/s3.go

```diff
@@ -208,6 +216,17 @@ func validate(conf Config) error {
 	if conf.AccessKey != "" && conf.SecretKey == "" {
 		return errors.New("no s3 secret_key specified while access_key is present in config file; either both should be present in config or envvars/IAM should be used.")
 	}
+
+	// ensure that the prefix isn't too long
+	if conf.Prefix != "" && len(conf.Prefix) >= 1024 {
+		return errors.New("prefix is too long (limited by Amazon at 1024 bytes long)")
+	}
+
+	// trim trailing slash and insert one
```

Comment:
Why are we doing this?

Reply:
normalization of paths for object keys, don't want two `/` on accident.

Reply:
Consider adding user validation, instead of trying to fix the error.

Reply:
Not sure I agree with that, I think this is a perfectly valid solution,